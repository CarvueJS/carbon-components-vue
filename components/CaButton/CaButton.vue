<template>
  <button data-button class="bx--btn bx--btn--primary" type="button">Primary button</button>
</template>

<script>
  import { FabButton } from 'carbon-components';

  export default {
    name: 'ca-button',
    mounted () {
      const button = FabButton.create(document.querySelector('[data-button]'));
    }
  }
</script>

<style></style>
