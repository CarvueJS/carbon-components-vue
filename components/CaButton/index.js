import CaButton from './CaButton.vue';

CaButton.install = Vue => {
  Vue.component(CaButton.name, CaButton);
}

export default CaButton
