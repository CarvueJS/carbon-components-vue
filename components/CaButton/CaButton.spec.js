import { shallow } from 'vue-test-utils';
import CaButton from './CaButton.vue';

describe('CaButton', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaButton);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
})
