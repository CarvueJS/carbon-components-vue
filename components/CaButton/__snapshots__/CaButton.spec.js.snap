// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaButton Mathes snapshot: danger 1`] = `
<button data-button="" type="button" class="bx--btn bx--btn--danger">
  <!---->
</button>
`;

exports[`CaButton Mathes snapshot: disabled 1`] = `
<button data-button="" type="button" disabled="disabled" class="bx--btn">
  <!---->
</button>
`;

exports[`CaButton Mathes snapshot: primary 1`] = `
<button data-button="" type="button" class="bx--btn bx--btn--primary">
  <!---->
</button>
`;

exports[`CaButton Mathes snapshot: secondary 1`] = `
<button data-button="" type="button" class="bx--btn bx--btn--secondary">
  <!---->
</button>
`;

exports[`CaButton Mathes snapshot: sm 1`] = `
<button data-button="" type="button" class="bx--btn bx--btn--sm">
  <!---->
</button>
`;
